### ISSUE TITLE
Use 'const' for final variables initialized to a constant value. Try replacing 'final' with 'const'.

### ISSUE THREAD

**Issue by liyofx** [trevorwang/retrofit.dart#652]
![image](https://github.com/trevorwang/retrofit.dart/assets/6176478/04cb4619-9786-448a-8269-bfad1e7576c1)

**Comment by emintolgahanpolat** [comment]
I think removing it from analysis is a better solution. Because having the code generator check these cases can cause a lot of confusion.

Warnings can be hidden from the analysis_options.yaml file.

`include: package:flutter_lints/flutter.yaml
analyzer:
  exclude:
    [
      build/**,
      lib/**.g.dart
    ]
linter:`

**Comment by JuKu** [comment]
@emintolgahanpolat But this doesn't solve the problem.
As I know, a const field is more performant, than a final field.

**Comment by emintolgahanpolat** [comment]
@JuKu  Yes you are right, a const field is more performant, than a final field.

But if _data wasn't null we couldn't use const. [const_initialized_with_non_constant_value](https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=const_initialized_with_non_constant_value#const_initialized_with_non_constant_value)

If _data is null, const can be used. Do you think this will solve the problem?

**Comment by emintolgahanpolat** [comment]
<img width="591" alt="image" src="https://github.com/trevorwang/retrofit.dart/assets/20768158/536d7621-6f13-447c-ac08-ae4550f747c8">

If you think this will solve the problem I can create a PR.

**Comment by JuKu** [comment]
@emintolgahanpolat I think this should work! :)
And it should solve the problem well.

Btw, I think these are the rules from the Flutter analyzer:
  - https://dart.dev/tools/linter-rules/prefer_const_constructors_in_immutables
  - https://dart.dev/tools/linter-rules/prefer_const_literals_to_create_immutables
  - https://dart.dev/tools/linter-rules/prefer_const_declarations

I think only the last rule is relevant for us, or?
